speaker-segmentation/seg_pyannote.py

```diff
@@ -73,3 +73,66 @@
         RttmSeq(segments).write(os.path.join(args.base_dir, "results",
                                              "pyannote", f"pa_ami_micro_test_sample_{i}.rttm"))
         print(f"Processed sample {i}: {filename}")
+"""
+
+# 13/05/2022 latest edit
+
+# function to filter out segments whose duration is below a threshold
+def duration_selection(segments, threshold):
+    segments_filtered = [seg for seg in segments
+       if seg.tdur > threshold] 
+    return segments_filtered
+
+# function to merge segments whose intersegment interval is below a threshold
+def merge_segments(segments, threshold, include_gaps = True):
+    merged_segments = [segments[0]]
+    for index, segment in enumerate(segments[1:]):
+        end = (merged_segments[-1].tbeg + merged_segments[-1].tdur) # end of first segment
+        gap_dur = (segment.tbeg - end)
+        merge = gap_dur <= threshold
+        if merge:
+            if include_gaps:  
+                joint_segment = RttmObj(
+                    type=segment.type,
+                    file=segment.file,
+                    chnl=segment.chnl,
+                    tbeg=float(merged_segments[-1].tbeg), #beginning of fist segment
+                    tdur=float(merged_segments[-1].tdur + gap_dur + segment.tdur) #duration 1st segment + gap + duration 2nd segment
+                )
+            else:
+                joint_segment = RttmObj(
+                    type=segment.type,
+                    file=segment.file,
+                    chnl=segment.chnl,
+                    tbeg=float(merged_segments[-1].tbeg), #beginning of fist segment
+                    tdur=float(merged_segments[-1].tdur + segment.tdur) #duration 1st segment + duration 2nd segment
+                )
+            merged_segments[-1] = joint_segment
+        else:
+            merged_segments.append(segment)
+    return merged_segments
+
+# threshold arguments
+filter_th = 0.5 # segment duration
+merge_th = 0.1 # segment gap 
+
+for i, sample in enumerate(data):
+    if i in file_indices:
+        filename = sample["file"]
+        result = pipeline(filename)
+        segments = [RttmObj(
+            type="SPEAKER",
+            file=filename,
+            chnl=1,
+            tbeg=float(seg.start),
+            tdur=float(seg.end - seg.start)
+        ) for seg in result.itersegments()]
+        
+        filtered_segments = duration_selection(segments, filter_th) # remove short segments
+        merged_segments = merge_segments(seg_filtered, merge_th) # merge close segments
```

Comment:
There is an variable name mismatch here.
```suggestion
        merged_segments = merge_segments(filtered_segments, merge_th) # merge close segments
```